llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8079,8 +8075,16 @@ EpilogueVectorizerEpilogueLoop::emitMinimumVectorEpilogueIterCountCheck(
 
   BranchInst &BI =
       *BranchInst::Create(Bypass, LoopVectorPreHeader, CheckMinIters);
-  if (hasBranchWeightMD(*OrigLoop->getLoopLatch()->getTerminator()))
-    setBranchWeights(BI, EpilogueMinItersBypassWeights);
+  if (hasBranchWeightMD(*OrigLoop->getLoopLatch()->getTerminator())) {
+    // Assume the tripcount for the epilogue loop is equally distributed between
+    // 0 and `VectorTripCount - 1` leaving us with a `VectorTripCount * 0.5`
+    // branch weight.
+    unsigned CombinedFactor = UF * VF.getKnownMinValue();
+    assert(CombinedFactor > 1 && "UF times VF should be bigger than one");
+    unsigned EstimatedTripCount = (CombinedFactor + 1) / 2;
+    uint32_t Weights[] = {1, EstimatedTripCount};
```

Comment:
Here we are computing the probability of (x < V) where x is the remaining loop trip count, and V is the remainder vector loop's step length. Assuming x is uniformly distributed between [0, V2] where V2 is UF*VF of the main loop, then the probability would be min(V, V2)/V2

Reply:
Thanks for catching this!